src/platform/packages/shared/response-ops/alerts-table/hooks/use_controllable_state.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import {
+  useCallback,
+  useRef,
+  useState,
+  type Dispatch,
+  type SetStateAction,
+  useEffect,
+} from 'react';
+import { isFunction } from 'lodash';
+
+type ChangeHandler<T> = (state: T) => void;
+type SetStateFn<T> = Dispatch<SetStateAction<T>>;
+
+export function useControllableState<T>(params: {
+  prop?: T;
+  onChange?: ChangeHandler<T>;
+  defaultValue?: T;
+}): [T, SetStateFn<T>];
+export function useControllableState<T>({
+  prop,
```

Comment:
nit: rename to propValue or just value because prop alone isn't clear enough what it's referring to, is it the prop name maybe?